errors.go

```diff
@@ -0,0 +1,56 @@
+package filclient
+
+import "fmt"
+
+type ErrorCode int
+
+const (
+	// Failed to connect to a miner.
+	ErrMinerConnectionFailed ErrorCode = iota
+
+	// There was an issue related to the Lotus API.
+	ErrLotusError
+)
+
+type Error struct {
+	Code  ErrorCode
+	Inner error
+}
+
+func (code ErrorCode) String() string {
+	var msg string
+
+	switch code {
+	case ErrMinerConnectionFailed:
+		msg = "miner connection failed"
```

Comment:
just return the message?

Reply:
LOL my mistake, it was originally mashed into another function and I did not use brain when I separated it. This is much easier, heh